### PR TITLE
Do not rely on Leader/Follower to position DropdownMenu menu

### LINKED ISSUE #157916: DropdownMenu menu always appears below the TextField
### Steps to reproduce

1. Run the code sample
2. Tap on the DropdownMenu

### Expected results

The menu appear above the text field as there is no space below:

image

### Actual results

The menu is not visible because it is positioned below the text field:

image

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  MyApp({super.key});

  final List<DropdownMenuEntry<String>> menuEntries =
      ["Red", "Green", "Blue"].map((t) => DropdownMenuEntry<String>(label: t, value: t)).toList();

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: Align(
          alignment: Alignment.bottomLeft,
          child: DropdownMenu<String>(
            requestFocusOnTap: true,
            dropdownMenuEntries: menuEntries,
          ),
        ),
      ),
    );
  }
}


```

</details>

### Screenshots or Video

_No response_

### Logs

_No response_

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel master, 3.27.0-1.0.pre.315, on Ubuntu 24.04.1 LTS 6.8.0-47-generic, locale fr_FR.UTF-8)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Chrome - develop for the web
[✓] Linux toolchain - develop for Linux desktop
[✓] Android Studio (version 2023.3)
[✓] VS Code (version 1.95.0)
[✓] Connected device (2 available)
[✓] Network resources```

</details>

### LINKED ISSUE #158924: [Regression] Cannot scroll to the bottom of `DropdownMenu`
### Steps to reproduce

Create a `DropdownMenu` with a long list on desktop. This behavior differs on the latest stable and main channels at the time of writing this. 

### Main channel

https://github.com/user-attachments/assets/4857b8d5-206b-4d73-8b7a-d3735bc61767

### Stable channel

https://github.com/user-attachments/assets/349d351d-68ce-43c0-92ae-d0f52e82b673

### Expected results

Can scroll to the bottom of `DropdownMenu`

### Actual results

Cannot reach the bottom of `DropdownMenu`

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      theme: ThemeData(useMaterial3: true),
      home: const CurveSelectorScreen(),
    );
  }
}

class CurveSelectorScreen extends StatefulWidget {
  const CurveSelectorScreen({super.key});

  @override
  State<CurveSelectorScreen> createState() => _CurveSelectorScreenState();
}

class _CurveSelectorScreenState extends State<CurveSelectorScreen> {
  Curve selectedCurve = Curves.bounceOut;

  final List<Curve> curves = [
    Curves.bounceOut,
    Curves.bounceIn,
    Curves.bounceInOut,
    Curves.easeIn,
    Curves.easeOut,
    Curves.easeInOut,
    Curves.linear,
    Curves.decelerate,
    Curves.elasticIn,
    Curves.elasticOut,
    Curves.elasticInOut,
    Curves.fastOutSlowIn,
    Curves.slowMiddle,
    Curves.ease,
    Curves.fastLinearToSlowEaseIn,
    Curves.decelerate
  ];

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: Text('Curve Selectors from ${curves.length}')),
      body: Center(
        child: DropdownMenu<Curve>(
          initialSelection: selectedCurve,
          onSelected: (value) {
            if (value != null) {
              setState(() {
                selectedCurve = value;
              });
            }
          },
          dropdownMenuEntries: curves.asMap().entries.map((entry) {
            final index = entry.key;
            final curve = entry.value;
            return DropdownMenuEntry<Curve>(
              value: curve,
              label: '$index - ${curve.toString()}',
            );
          }).toList(),
        ),
      ),
    );
  }
}

```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[!] Flutter (Channel main, 3.27.0-1.0.pre.494, on macOS 15.1 24B83 darwin-arm64,
    locale en-US)
    • Flutter version 3.27.0-1.0.pre.494 on channel main at
      /Users/tahatesser/Code/flutter
    ! Upstream repository git@github.com:TahaTesser/flutter.git is not the same
      as FLUTTER_GIT_URL
    • FLUTTER_GIT_URL = git@github.com:TahaTesser/flutter.gitexport
    • Framework revision 8f62a9904d (3 hours ago), 2024-11-14 05:39:39 -0500
    • Engine revision 619804c0fb
    • Dart version 3.7.0 (build 3.7.0-140.0.dev)
    • DevTools version 2.41.0-dev.2
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

```

</details>

### PR DESCRIPTION
**Pull request by bleroux**
## Description

This PR removes `DropdownMenu` usage of Leader/Follower.

Leader/Follower positioning was introduced in https://github.com/flutter/flutter/pull/154667 which uses Leader/Follower approach to fix some weird positioning issues (such as https://github.com/flutter/flutter/issues/149037).

Unfortunately, it also introduces some regressions, see:
- https://github.com/flutter/flutter/issues/157916
- https://github.com/flutter/flutter/issues/158924

Because https://github.com/flutter/flutter/pull/154667 is already included in the beta channel, cherry-picking this PR should be considered.

## Context

This PR is not a full revert and keeps Leader/Follower usage in `MenuAnchor` because this usage is optional and doesn't cause any regression.
There are some ongoing work which might fix or mitigate this problem:
- https://github.com/flutter/flutter/pull/157921
- https://github.com/flutter/flutter/pull/158255

## Related Issue

Fixes https://github.com/flutter/flutter/issues/157916
Fixes https://github.com/flutter/flutter/issues/158924

Reopens https://github.com/flutter/flutter/issues/123395
Reopens https://github.com/flutter/flutter/issues/149037
Reopens https://github.com/flutter/flutter/issues/151856

### CODE REVIEW

**Comment by gspencergoog** [comment]
Seems like there is a tradeoff here, and we need to decide which set of bugs is more detrimental to the user.

The regressions look like they are annoying, but the issues that this reopens are also pretty bad: they hide the dropdown on a mobile device because the keyboard covers up the menu, and the menu gets disconnected from the anchor when scrolling.

Are you expecting that the two other PRs referenced will fix the reopened issues, or are they just related?

Also, the "Fixes" section in this PR description seem to refer to the PRs, not the bugs, so you might want to update the numbers there.

**Comment by bleroux** [comment]
Thanks for the feedback :pray: 

(I have edited the description to fix the two links in the "fixes" section, thank for noticing this).

> Seems like there is a tradeoff here, and we need to decide which set of bugs is more detrimental to the user.

Yes, there is a tradeoff. The main reason I filed this PR is that we are probably close to a stable release and I think this has an impact on the decision.

> Are you expecting that the two other PRs referenced will fix the reopened issues, or are they just related?

[One of the PR,](https://github.com/flutter/flutter/pull/157921), is a direct attempt to fix [the first regression](https://github.com/flutter/flutter/issues/157916). But it is still a draft and It might take time to discuss and review this solution and it probably needs more work because being able to move the menu above the field is a good first step but not as smart as the current positioning logic that MenuAnchor offers when not using Leader/Follower.

The second PR introduces RawMenuAnchor in the widget library, I don't think it will fix the reopened issues. But it might help. It is a promising and huge PR so it might takes time to review it.

In summary, there are pending work that will help to fix the referenced regressions, but the corresponding PRs might not be ready for the next stable release.

**Comment by justinmc** [comment]
### Option 1: Close this PR

Downside: We'll be stuck with the following two issues. What would be the long term plan for addressing these if we go with this option?

https://github.com/flutter/flutter/issues/157916: DropdownMenu menu always appears below the TextField
A bad one because Dropdowns at or near the bottom of the screen are unusable on all platforms. The workaround would be to move the Dropdown.

https://github.com/flutter/flutter/issues/158924: Cannot scroll to the bottom of DropdownMenu
Also a bad one because latter menu items are not clickable. It seems to affect all platforms. Affects only longer lists,  but that's probably common (in the issue, a Dropdown in the center of the screen with 8 items repros the bug). Workaround would just be to keep the list shorter.

### Option 2: Merge this PR

The above two issues will be fixed, but the following ones will be reopened. They might be fixed with work in progress in https://github.com/flutter/flutter/pull/157921 and https://github.com/flutter/flutter/issues/158924, but those probably won't be done by the next release.

https://github.com/flutter/flutter/issues/123395: Menu appears behind the keyboard
Android only, doesn't happen all the time. The user can work around the problem by closing the keyboard. Admittedly though, it looks very bad.

https://github.com/flutter/flutter/issues/149037: DropdownMenu Entries Not Positioned Correctly on Keyboard Appearance
Looks to me like a duplicate of the previous issue.

https://github.com/flutter/flutter/issues/151856: DropdownMenu doesn't follow in scrollable
All platforms. Affects any Dropdown in a Scrollable, which is probably a good amount. The user can work around the problem by closing and reopening the menu and not scrolling. Looks bad.

## So what should we do?

It looks to me like option 2 gets us in a better state. The issues we'd have are ugly but not app-breaking. Also, we seem to have a potential path forward to fixing them with https://github.com/flutter/flutter/pull/157921 and https://github.com/flutter/flutter/issues/158924. I say we merge this PR and then follow up on making sure those issues get fixed.

**Review by justinmc — APPROVED** [review]
LGTM 👍 

See my thoughts in the previous comment. I'm definitely open to being corrected but I think merging this is the best of two imperfect options.

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

Works for me.  I just wanted it to be a reasoned choice.

**Comment by justinmc** [comment]
@itsjustkevin The two PRs that this reverts (https://github.com/flutter/flutter/issues/157916 and https://github.com/flutter/flutter/issues/158924) have already rolled into beta, so this PR should be cherry picked into beta if possible.
